### DISCUSSION TITLE
Progress bar for fori_loop

### DISCUSSION THREAD

**Discussion by rezaaskary** [jax-ml/jax#13126] [Ideas] [answered]
Hi,

does anyone know an efficient way to add progress bar to lax.fori_loop? something like integrating tqdm with for loop.

**Comment by sharadmv** [comment, marked as answer]
I created an example of you how can use `jax.debug.callback` to create a progress bar system for JAX:

https://gist.github.com/sharadmv/be24b3107cf9b8bf027ea8e2f177882e

Does this work?

**Reply by rezaaskary** [reply]
That's great! thank you
